target/linux/bmips/dts/bcm63168-sagem-fast-3864-op.dts

```diff
@@ -0,0 +1,310 @@
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ethernet {
+	status = "okay";
+
+	nvmem-cells = <&macaddr_cferom_6a0>;
+	nvmem-cell-names = "mac-address";
+};
+
+&leds {
+	status = "okay";
+	brcm,serial-leds;
+	brcm,serial-dat-low;
+	brcm,serial-shift-inv;
+	brcm,serial-mux;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_leds &pinctrl_serial_led>;
+
+	led@0 {
+		reg = <0>;
+		active-low;
+		function = LED_FUNCTION_WPS;
+		color = <LED_COLOR_ID_GREEN>;
+	};
+
+	led@1 {
+		reg = <1>;
+		active-low;
+		function = LED_FUNCTION_WPS;
+		color = <LED_COLOR_ID_RED>;
+	};
+
+	led@2 {
+		reg = <2>;
+		active-low;
+		label = "red:internet";
+	};
+
+	led_dsl_green: led@3 {
+		reg = <3>;
+		active-low;
+		label = "green:dsl";
+	};
+
+	led@4 {
+		reg = <4>;
+		active-low;
+		label = "green:fxs";
+	};
+
+	led@5 {
+		reg = <5>;
+		active-low;
+		label = "red:fxs";
+	};
+
+	led@6 {
+		reg = <6>;
+		active-low;
+		label = "green:dummy6";
+	};
+
+	led@7 {
+		reg = <7>;
+		active-low;
+		label = "green:dummy7";
+	};
+
+	led_internet_green: led@8 {
+		reg = <8>;
+		active-low;
+		label = "green:internet";
+	};
+
+	led_usb_green: led@9 {
+		reg = <9>;
+		active-low;
+		label = "green:dsl_bonding";
+	};
+
+	led_power_red: led@15 {
+		reg = <15>;
+		active-low;
+		function = LED_FUNCTION_POWER;
+		color = <LED_COLOR_ID_RED>;
+	};
+
+	led_power_green: led@20 {
+		reg = <20>;
+		active-low;
+		default-state = "on";
+		function = LED_FUNCTION_POWER;
+		color = <LED_COLOR_ID_GREEN>;
+	};
+};
+
+&mdio_ext {
+	switch@1e {
+		compatible = "brcm,bcm53125";
+		reg = <0x1e>;
+
+		dsa,member = <1 0>;
+
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@4 {
+				reg = <4>;
+				label = "lan1";
+			};
+
+			port@3 {
+				reg = <3>;
+				label = "lan2";
+			};
+
+			port@2 {
+				reg = <2>;
+				label = "lan3";
+			};
+
+			port@1 {
+				reg = <1>;
+				label = "lan4";
+			};
+
+			port@8 {
+				reg = <8>;
+
+				phy-mode = "rgmii";
+				ethernet = <&switch0port4>;
+
+				fixed-link {
+					speed = <1000>;
+					full-duplex;
+					asym-pause;
+					pause;
+				};
+			};
+		};
+	};
+};
+
+&nflash {
+	status = "okay";
+
+	nandcs@0 {
+		compatible = "brcm,nandcs";
+		reg = <0>;
+		nand-ecc-step-size = <512>;
+		nand-ecc-strength = <15>;
+		nand-on-flash-bbt;
+		brcm,nand-oob-sector-size = <64>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "cferom_nvram";
+				reg = <0x00000000 0x00020000>;
+				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					macaddr_cferom_6a0: macaddr@6a0 {
+						reg = <0x6a0 0x6>;
+					};
+				};
+			};
+
+			partition@20000 {
+				label = "stock_rootfs";
+				reg = <0x00020000 0x3d60000>;
+				read-only;
+			};
+
+			partition@3d80000 {
+				label = "stock_rootfs_update";
+				reg = <0x03d80000 0x3d60000>;
+			};
```

Comment:
Why are we exposing this?
These are the same as wfi partition defined below:
```
			partition {
				compatible = "brcm,wfi-split";
				label = "wfi";
				reg = <0x00020000 0x7ac0000>;
			};
```
I think that we should remove these two and move the "brcm,wfi-split" partition here.

Reply:
They would allow the user to dump the stock firmware when booting in [the RAM mode](https://openwrt.org/inbox/toh/sagem/f_st3864op#boot_openwrt_in_ram).

Reply:
> They would allow the user to dump the stock firmware when booting in [the RAM mode](https://openwrt.org/inbox/toh/sagem/f_st3864op#boot_openwrt_in_ram).

Users can still do that by dumping the whole wfi partition or the split partitions.